### PR TITLE
Review me! I am done with my room and test suite 

### PR DESCRIPTION
**Pull request by griffpete**
I put a warning on line 222, i would recommend reading. Basically the slot machine and black jack features where taken from github then vibe coded.

 Reviewing the top half of my room code and test suite would be a better use of time :)

### CODE REVIEW

File: Room15_griff.py [hunk 1]
```diff
@@ -669,15 +655,10 @@ def calculate_payout(reels, wager, current_score):
             elif symbol == 'O':
                 payouts.append(('MEGA JACKPOT - ALL Os!!!', wager * 500))
                 winning_lines.append((positions, symbol))
-            # For all Ls, we already handled it above
-            return payouts, winning_lines  # Mega jackpot overrides all other payouts
+            return payouts, winning_line
```
Comment (OliverOlsenUT) [inline Room15_griff.py]: `winning_line` doesn't exist, this will crash the code

File: Room15_griff.py [hunk 2]
```diff
@@ -79,13 +79,13 @@ def describe_room(self):
 
     def move(self, direction):
         if direction in ["east", "e"]:
-            print("You head through the door to the east.")
+            print("You take the east exit out of the casino.")
```
Comment (OliverOlsenUT) [inline Room15_griff.py]: there's no exit to the east

**Review by OliverOlsenUT — CHANGES_REQUESTED** [review]
You can have 0 health after betting on bloodjack. This doesn't kill you.
# What items are done particularly well in the code?
Betting health and points is very cool. Using variables for each color is good, it makes the code more readable.
# What suggestions would you make to improve the code? (be inclusive and very specific)
(Note that these are suggestions, you don't have to implement these changes)
The unnecessary imports could be removed to clean up the code a little.
In `look()`, there's unreachable code (this originates from the room you based this room on):
```py
        if target == item.name.lower().strip():
            print(item.description)
            if(item.state != None):
                print(f"The {item.name} is {item.state}")
            return
            # unreachable
            print("looking at", target, "reveals nothing.")
            return
```

# Overall rating
How well is the code organized: **Great**
Consistent and clear variable names: **Great**
How well did the code meet the requirements: **Great**
How testable is the code: **Needs work**
Unit tests should check for some edge cases, like the bloodjack minigame killing you when you're at 0hp.

File: Room15_griff.py (hunk 2 above)
Comment (griffpete) [reply Room15_griff.py]: I changed east to west, good catch!

**Comment by griffpete** [comment]
"You can have 0 health after betting on bloodjack. This doesn't kill you.", I am updating the health variable to 0, but it does not call this line in Adventure.py
``` python
 if not player.is_alive():
   print("You have perished in the dungeon! Game over. Your score is: ", player.score)
   return
```

Any suggestions?

**Comment by OliverOlsenUT** [comment]
For my room I check if the player is alive every time I update the health variable.
```py
def damage_player(self, player, amount):
        player.health -= amount
        if not player.is_alive():
            print("You have perished in the dungeon! Game over. Your score is:", player.score)
            sys.exit(0)
```
